test/play_scraper_test.py

```diff
@@ -34,3 +43,38 @@ def test_get_category(self, play_scraper, test_dir):
         genre = play_scraper.get_category("com.example.bis")
 
         assert expected_genre == genre
+
+    def test_missing_icon_class(self, play_scraper, test_dir):
+        html_file = test_dir / "com.class.error"
+        html_file.write_text(SIMPLE_HTML)
+
+        with pytest.raises(PlayScrapingError) as exc:
+            play_scraper.get_icon("com.class.error", ICON_SUBDIR)
+
+        assert str(exc.value) == "Icon class not found in html."
+
+    def test_missing_category_itemprop(self, play_scraper, test_dir):
+        html_file = test_dir / "com.itemprop.error"
+        html_file.write_text(SIMPLE_HTML)
+
+        with pytest.raises(PlayScrapingError) as exc:
+            play_scraper.get_category("com.itemprop.error")
+
+        assert str(exc.value) == "Category itemprop not found in html."
+
+    def test_invalid_base_url(self, icon_dir):
+        random_int = np.random.randint(10000)
+        random_url = "https://127.0.0.1/" + str(random_int) + "/"
```

Comment:
do you need rand here? :) Just manually hardcode something

Reply:
Also 10000 is too small and could still cause a collision. I would use something like `betel-test-invalid-base-url` to make it specific to your codebase which makes collisions much less likely. If you want to be extra sure `betel-test-invalid-base-url-938TD`